src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -855,7 +868,21 @@ def _find_using_common_tenant(self, access_token, resource):
             subscriptions = self._find_using_specific_tenant(
                 tenant_id,
                 temp_credentials[_ACCESS_TOKEN])
-            all_subscriptions.extend(subscriptions)
+
+            # When a subscription can be listed by multiple tenants, only the first appearance is retained
+            for sub_to_add in subscriptions:
+                add_sub = True
+                for sub_to_compare in all_subscriptions:
```

Comment:
It looks that `all_subscriptions` in your current code is [].

Reply:
`all_subscriptions` will be filled at L885 in the first iteration.